plugins/eddn.py

```diff
@@ -325,96 +408,159 @@ def canonicalise(self, item):
         for k, v in d.items():
             if k.endswith('_Localised'):
                 pass
-            elif hasattr(v, 'items'):	# dict -> recurse
+
+            elif hasattr(v, 'items'):  # dict -> recurse
                 filtered[k] = filter_localised(v)
-            elif isinstance(v, list):	# list of dicts -> recurse
+
+            elif isinstance(v, list):  # list of dicts -> recurse
                 filtered[k] = [filter_localised(x) if hasattr(x, 'items') else x for x in v]
+
             else:
                 filtered[k] = v
+
         return filtered
 
     # Track location
-    if entry['event'] in ['Location', 'FSDJump', 'Docked', 'CarrierJump']:
+    if entry['event'] in ('Location', 'FSDJump', 'Docked', 'CarrierJump'):
         if entry['event'] in ('Location', 'CarrierJump'):
-            this.planet = entry.get('Body') if entry.get('BodyType') == 'Planet' else None
+            this.planet: Optional[str] = entry.get('Body') if entry.get('BodyType') == 'Planet' else None
+
         elif entry['event'] == 'FSDJump':
-            this.planet = None
+            this.planet: Optional[str] = None
+
         if 'StarPos' in entry:
-            this.coordinates = tuple(entry['StarPos'])
+            this.coordinates: Optional[Tuple[int, int, int]] = tuple(entry['StarPos'])
+
         elif this.systemaddress != entry.get('SystemAddress'):
-            this.coordinates = None	# Docked event doesn't include coordinates
-        this.systemaddress = entry.get('SystemAddress')
+            this.coordinates: Optional[Tuple[int, int, int]] = None  # Docked event doesn't include coordinates
+
+        this.systemaddress: Optional[str] = entry.get('SystemAddress')
+
     elif entry['event'] == 'ApproachBody':
         this.planet = entry['Body']
-    elif entry['event'] in ['LeaveBody', 'SupercruiseEntry']:
+
+    elif entry['event'] in ('LeaveBody', 'SupercruiseEntry'):
         this.planet = None
 
     # Send interesting events to EDDN, but not when on a crew
     if (config.getint('output') & config.OUT_SYS_EDDN and not state['Captain'] and
         (entry['event'] in ('Location', 'FSDJump', 'Docked', 'Scan', 'SAASignalsFound', 'CarrierJump')) and
-        ('StarPos' in entry or this.coordinates)):
+            ('StarPos' in entry or this.coordinates)):
+
         # strip out properties disallowed by the schema
-        for thing in ['ActiveFine', 'CockpitBreach', 'BoostUsed', 'FuelLevel', 'FuelUsed', 'JumpDist', 'Latitude', 'Longitude', 'Wanted']:
+        for thing in (
+            'ActiveFine',
+            'CockpitBreach',
+            'BoostUsed',
+            'FuelLevel',
+            'FuelUsed',
+            'JumpDist',
+            'Latitude',
+            'Longitude',
+            'Wanted'
+        ):
             entry.pop(thing, None)
+
         if 'Factions' in entry:
             # Filter faction state. `entry` is a shallow copy so replace 'Factions' value rather than modify in-place.
```

Comment:
Mention in the comment that this is to comply with schema restrictions not allowing personal data over EDDN.